### DISCUSSION TITLE
How to tell once the pdf is fully loaded along with html?

### DISCUSSION THREAD

**Discussion by chronicadventure** [TaTo30/vue-pdf#163] [Q&A] [answered]
I'm doing my own highlighting in the background as I ran into issues in an older version where highlighting would not work do to the way spans were broken up. So essentially I'm using LLMs to read the rendered html and then return which spans I should add the highlight class to. I know it's not efficient but I'm having trouble figuring out once the html spans have been added to the dom. 

I tried using an observer to observe the <body> tag for any nodes that are added with span[role="presentation"] but that is not working

**Comment by TaTo30** [comment, marked as answer]
You could use the `text-loaded` event to catch up when the text has added to the dom, this event even returns the list of span elements added at the text layer. Ref: https://tato30.github.io/vue-pdf/examples/loaded_events/text_loaded.html

**Reply by chronicadventure** [reply]
I was hoping for an easy solution like this. Thanks!
